test/e2e/storage/testsuites/snapshottable.go

```diff
@@ -282,6 +297,58 @@ func (s *snapshottableTestSuite) DefineTests(driver storageframework.TestDriver,
 					framework.ExpectError(err)
 				}
 			})
+
+			ginkgo.It("start pod, check the claim, check pv, create snapshot, create new pvc from snapshot, new pod to use the pvc (with initial pod attached), validate data on volume", func() {
+				createAndTestPodClaimPV(true)
+				createSnapshot()
+
+				ginkgo.By("checking the snapshot")
+				// Get new copy of the snapshot
+				vs, err = dc.Resource(storageutils.SnapshotGVR).Namespace(vs.GetNamespace()).Get(context.TODO(), vs.GetName(), metav1.GetOptions{})
+				framework.ExpectNoError(err)
+
+				var secondPVC *v1.PersistentVolumeClaim
+				var secondPod *v1.Pod
+
+				ginkgo.By("creating a pvc from the snapshot")
+				secondPVC = e2epv.MakePersistentVolumeClaim(e2epv.PersistentVolumeClaimConfig{
+					ClaimSize:        claimSize,
+					StorageClassName: &(sc.Name),
+				}, config.Framework.Namespace.Name)
+
+				group := "snapshot.storage.k8s.io"
+
+				secondPVC.Spec.DataSource = &v1.TypedLocalObjectReference{
+					APIGroup: &group,
+					Kind:     "VolumeSnapshot",
+					Name:     vs.GetName(),
+				}
+
+				secondPVC, err = cs.CoreV1().PersistentVolumeClaims(secondPVC.Namespace).Create(context.TODO(), secondPVC, metav1.CreateOptions{})
+				framework.ExpectNoError(err)
+				cleanupSteps = append(cleanupSteps, func() {
+					framework.Logf("deleting claim %q/%q", secondPVC.Namespace, secondPVC.Name)
+					// typically this claim has already been deleted
+					err = cs.CoreV1().PersistentVolumeClaims(secondPVC.Namespace).Delete(context.TODO(), secondPVC.Name, metav1.DeleteOptions{})
+					if err != nil && !apierrors.IsNotFound(err) {
+						framework.Failf("Error deleting claim %q. Error: %v", secondPVC.Name, err)
+					}
+				})
+
+				ginkgo.By("starting pod2 to use the claim on the same node")
+
+				secondPod = StartInPodWithVolume(cs, secondPVC.Namespace, secondPVC.Name, "second-pvc-tester", "sleep 300", config.ClientNodeSelection)
```

Comment:
This isn't actually the same node, it's the same topology. If you actually want the same node, you can follow what we did for the multivolume tests: https://github.com/kubernetes/kubernetes/blob/035f4328f062f0fae2c2d2647e0c36c51304b538/test/e2e/storage/testsuites/multivolume.go#L459

It may also be worthwhile to have another test case with a different node.

Reply:
Thanks, I've updated and also added another test case with a different node.